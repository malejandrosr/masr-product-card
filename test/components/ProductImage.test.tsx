import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('Pruebas en <ProductImage />', () => {
	it('Debe mostrar el componente correctamente con una imagen personalizada', () => {
		const wrapper = renderer.create(
			<ProductImage img="./image.png" className="custom-class" />
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('Debe de mostrar el componente con la imagen del producto', () => {
		const wrapper = renderer.create(
			<ProductCard product={product2}>
				{() => (
					<>
						<ProductImage />
					</>
				)}
			</ProductCard>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
