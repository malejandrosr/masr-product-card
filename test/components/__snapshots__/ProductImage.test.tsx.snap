// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductImage /> Debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Pic"
    className="productImg undefined"
    src="coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en <ProductImage /> Debe mostrar el componente correctamente con una imagen personalizada 1`] = `
<img
  alt="Product Pic"
  className="productImg custom-class"
  src="./image.png"
/>
`;
